src/devices/Ft232H/Ft232HDevice.cs

```diff
@@ -0,0 +1,857 @@
+        }
+
+        #endregion
+
+        #region SPI
+
+        internal void SpiInitialize()
+        {
+            // Do we already have SPI setup?
+            if (IsSpiMode)
+            {
+                // No need to initialize everything
+                return;
+            }
+
+            GetHandle();
+            IsSpiMode = true;
+            var ftStatus = FtFunction.FT_SetLatencyTimer(_ftHandle, 1);
+            ftStatus = FtFunction.FT_SetUSBParameters(_ftHandle, 65535, 65535);
+            ftStatus = FtFunction.FT_SetChars(_ftHandle, 0, 0, 0, 0);
+            ftStatus = FtFunction.FT_SetTimeouts(_ftHandle, 3000, 3000);
+            ftStatus = FtFunction.FT_SetLatencyTimer(_ftHandle, 1);
+            // Reset
+            ftStatus = FtFunction.FT_SetBitMode(_ftHandle, 0x00, 0x00);
+            // Enable MPSSE mode
+            ftStatus = FtFunction.FT_SetBitMode(_ftHandle, 0x00, 0x02);
+            if (ftStatus != FtStatus.Ok)
+            {
+                throw new IOException($"Failed to setup device {Description}, status: {ftStatus} in MPSSE mode");
+            }
+
+            // 50 ms according to thr doc for all USB to complete
+            Thread.Sleep(50);
+            FlushBuffer();
+            SetupMpsseMode();
+
+            int idx = 0;
+            Span<byte> toSend = stackalloc byte[10];
+            toSend[idx++] = (byte)FtOpcode.DisableClockDivideBy5;
+            toSend[idx++] = (byte)FtOpcode.TurnOffAdaptativeClocking;
+            toSend[idx++] = (byte)FtOpcode.Disable3PhaseDataClocking;
+            toSend[idx++] = (byte)FtOpcode.SetDataBitsLowByte;
+            // Pin clock output, MISO output, MOSI input
+            _gpioLowDir = (byte)((_gpioLowDir & 0xF8) | 0x03);
+            // clock, MOSI and MISO to 0
+            _gpioLowData = (byte)(_gpioLowData & 0xF8);
+            toSend[idx++] = _gpioLowDir;
+            toSend[idx++] = _gpioLowData;
+            // The SK clock frequency can be worked out by below algorithm with divide by 5 set as off
+            // TCK period = 60MHz / (( 1 + [ (0xValueH * 256) OR 0xValueL] ) * 2)
+            // Command to set clock divisor
+            toSend[idx++] = (byte)FtOpcode.SetClockDivisor;
+            uint clockDivisor = (uint)(60000 / ((_connectionSettings[0].ClockFrequency / 1000) * 2) - 1);
+            toSend[idx++] = (byte)(clockDivisor & 0xFF);
+            toSend[idx++] = (byte)(clockDivisor >> 8);
+            // loopback off
+            toSend[idx++] = (byte)FtOpcode.DisconnectTDItoTDOforLoopback;
+            Write(toSend);
+            // Delay as in the documentation
+            Thread.Sleep(30);
+        }
+
+        internal void SpiDeinitialize()
+        {
+            if (_connectionSettings.Count == 0)
+            {
+                IsSpiMode = false;
+            }
+        }
+
+        internal void SpiWrite(SpiConnectionSettings settings, ReadOnlySpan<byte> buffer)
+        {
+            if (buffer.Length > 65535)
```

Comment:
do all the pre-checks need to happen on all writes? should settings be part of the initialization instead?

Reply:
settings is the easy way to get the chipselect and the default state of the chip select pin. You can have as many SpiDevice as you want, just limited by the number of GPIO available. So a maximum of 13. 
Now for the size of the buffer, I have to say I was mazy. I can do a loop and send the sliced buffer as many times as needed. Will consider to add this. This will make less checks